models/metrics.go

```diff
@@ -306,3 +307,83 @@ func (m *Metrics) SelectQueryExamples(ctx context.Context, from, to time.Time, f
 	}
 	return &res, nil
 }
+
+const queryObjectDetailsLabelsTmpl = `
+	SELECT d_server, d_database, d_schema, d_username, d_client_host, labels.key AS lkey, labels.value AS lvalue
+	  FROM metrics
+	 ARRAY JOIN labels
+	 WHERE period_start >= :from AND period_start <= :to
+	 {{ if index . "filter" }} AND {{ index . "group" }} = :filter {{ end }}
+	 ORDER BY d_server, d_database, d_schema, d_username, d_client_host, labels.key, labels.value
+`
+
+// SelectObjectDetailsLabels selects object details labels for given time range and object.
+func (m *Metrics) SelectObjectDetailsLabels(ctx context.Context, from, to time.Time, filter,
+	group string) (*qanpb.ObjectDetailsLabelsReply, error) {
+	arg := map[string]interface{}{
+		"from":   from,
+		"to":     to,
+		"filter": filter,
+		"group":  group,
+	}
+	var queryBuffer bytes.Buffer
+	if tmpl, err := template.New("queryObjectDetailsLabelsTmpl").Funcs(funcMap).Parse(queryObjectDetailsLabelsTmpl); err != nil {
+		log.Fatalln(err)
```

Comment:
https://golang.org/pkg/text/template/#Must

It is also better to move it to the package-level variable:
```go
var queryObjectDetailsLabelsTmpl = template.Must(template.New("").Funcs(…).Parse(…))
```
That way it will panic during startup, not when `SelectObjectDetailsLabels` is called.

Reply:
Moved.